srec_cat/arglex3.h

```diff
@@ -57,17 +57,17 @@ class srec_cat_arglex3:
     /**
       * The default constructor.  Do not use.
       */
-    srec_cat_arglex3();
+    srec_cat_arglex3() = delete;
```

Comment:
It's the explicitness: You read the declaration and immediately know, that there won't be an implementation.
Static code analyzers would otherwise also warn, that there is a declaration without implementation.

For C++98 I would change the code and add an inline: If someone tries to use it, the compiler will error instead of linker.

Reply:
Fair call. Let's keep it in!